src/reuse/header.py

```diff
@@ -351,19 +351,26 @@ def _verify_paths_line_handling(
 
 
 def _verify_paths_comment_style(paths: List[Path], parser: ArgumentParser):
+    # Create empty list for files whose comment style is unknown
+    unrecognised_files = []
+
     for path in paths:
         style = _get_comment_style(path)
         not_uncommentable = not _is_uncommentable(path)
 
         # TODO: This check is duplicated.
         if style is None and not_uncommentable:
-            parser.error(
-                _(
-                    "'{path}' does not have a recognised file extension,"
-                    " please use --style, --explicit-license or"
-                    " --skip-unrecognised"
-                ).format(path=path)
-            )
+            unrecognised_files.append(path)
+
+    # If at least one file was unrecognised, print error
+    if len(unrecognised_files) > 0:
+        parser.error(
+            _(
+                "The following files do not have a recognised file extension."
+                " No file has been edited. Please use --style,"
+                " --explicit-license or --skip-unrecognised:\n{paths}"
```

Comment:
I'm not entirely happy with `\n{paths}` in the translatable string. If it weren't a translatable string, it'd be fine, but it's a good idea to not put formatting in translatable strings if you can help it.